streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -92,7 +92,9 @@ public void resume() {
     public void commit() {
         log.trace("{} Committing", logPrefix);
         stateMgr.flush();
-        stateMgr.checkpoint(Collections.<TopicPartition, Long>emptyMap());
+        if (!eosEnabled) {
```

Comment:
We write the checkpoint file when closing the `StateManager` -- so this is fine. Nevertheless, I did double check the current behavior and made some minor changes. Also updated https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Streams+Architecture

The code should align to this design (please also verify the design).